codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java

```diff
@@ -809,6 +813,7 @@ private void lastKnownStream(int lastKnownStream) {
             if (!alreadyNotified) {
                 notifyGoingAway();
             }
+            closeStreamsGreaterThanLastKnownStream();
```

Comment:
@buchgr it seems reasonable although I haven't though of all of the side-effects :)